src/main/java/com/google/devtools/build/lib/vfs/WindowsOsPathPolicy.java

```diff
@@ -52,10 +56,21 @@ public String resolveShortPath(String path) {
     }
   }
 
-  WindowsOsPathPolicy() {
-    this(new DefaultShortPathResolver());
+  static class CrossPlatformShortPathResolver implements ShortPathResolver {
+    @Override
+    public String resolveShortPath(String path) {
+      // Short paths can only be resolved on a Windows host.
+      // Skipping short path resolution when running on a non-Windows host can
+      // result in paths considered different that are actually the same.
+      // TODO: Consider failing when a short path is detected on a non-Windows
+      //  host. Since short path segments can arise from most operations on
+      //  PathFragment, this would however require exception handling in many
+      //  places.
```

Comment:
Is this something that you would want to do in the scope of this PR, or do you think it should be done separately?
Also, is this introducing a new failure mode (skipping short path resolution when running on a non-Windows host), or is this something that is also a problem today, without your change?

Reply:
I don't think it will ever be done given the effort it would take (essentially all `PathFragment` methods would gain a checked exception). It trades a real issue that affects Bazel today (paths are normalized incorrectly in cross-platform builds, which makes RBE almost impossible) for a highly theoretical issue (say, a conflict or a filesystem race between two actions that differ only in that one uses a short path and one doesn't).